src/mgmt/workflows/chains/create.py

```diff
@@ -6,13 +6,23 @@
 import logging
 from common.workflow import *
 from operators.chains_operator import *
+from operators.dfts_operator import *
+from operators.ftns_operator import *
 
 logger = logging.getLogger()
 chains_opr = ChainOperator()
+dfts_opr = DftOperator()
+ftns_opr = FtnOperator()
 
 
-def chain_create(chain, name, body, spec):
+def chain_create(task, chain, name, body, spec):
     logger.info("Creating Chain {}!".format(name))
     if not chain:
         chain = chains_opr.get_stored_obj(name, spec)
+    if not dfts_opr.store.contains_obj(chain.dft, KIND.dft):
+        task.raise_temporary_error(
+            "Parent DFT {} not yet created!".format(chain.dft))
+    ftns_opr.create_n_ftns(chain)
+    chain.tail = chain.ftns[chain.size - 1]
+    chain.head = chain.ftns[0]
     return chain
```

Comment:
Should have number of chains and replicas decided based on number of ftn droplets

Reply:
My thought was to have the cluster admin decide on the number of chains and replicas/chain based on the DFT object spec.
From DFT crd definition
```
    - name: Numchains
      type: integer
      priority: 0
      JSONPath: .spec.numchains
      description: The number of chains in the DFT
    - name: Numchainreplicas
      type: integer
      priority: 0
      JSONPath: .spec.numchainreplicas
      description: The number of replica ftns in each chain
```
Zeta can then give a temporary error and retry until the number of required droplets are available.

If we are to have these dynamically allocated based on existing resources, we can remove these fields in the DFT crd definition, and instead set a limit on the minimum number of droplets that's needed for a DFT to be created.
What are your thoughts?